auth-service/src/test/java/com/auth/authservice/presentationlayer/User/UserControllerIntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+package com.auth.authservice.presentationlayer.User;
+
+import com.auth.authservice.businesslayer.UserService;
+import com.auth.authservice.datalayer.user.User;
+import com.auth.authservice.datalayer.user.UserRepo;
+import com.auth.authservice.security.JwtTokenUtil;
+import org.aspectj.lang.annotation.Before;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import java.util.List;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@AutoConfigureWebTestClient
+class UserControllerIntegrationTest {
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private UserRepo userRepo;
+
+
+    @Autowired
+    private JwtTokenUtil jwtTokenUtil;
+
+    @Before("setup")
+    public void setup() {
+        String baseUri = "http://localhost:" + "9200";
+        this.webTestClient = WebTestClient.bindToServer().baseUrl(baseUri).build();
+    }
+
+    @Test
+    void validateToken_ShouldSucceed() {
+
+        List<User> users = userRepo.findAll();
+
+        User user = users.get(0);
+
+        String token = jwtTokenUtil.generateToken(user);
+
+        webTestClient.post()
+                .uri("/users/validate-token")
+                .accept(MediaType.APPLICATION_JSON)
+                .cookie("Bearer", token)
+                .bodyValue(token)
+                .exchange()
+                .expectStatus().isOk();
+
+
+
+    }
+
+
+
```

Comment:
This is for the entire test class but your spacing is kind crazy here try not to add too many unnecessary spaces